src/track/serato/markers2.cpp

```diff
@@ -391,5 +393,69 @@ QByteArray SeratoMarkers2::data() const {
     return outerData.leftJustified(size, '\0');
 }
 
+void SeratoMarkers2::syncToTrackObject(Track* track) const {
+    qDebug() << "Syncing 'Serato Markers2' tag data to Track object...";
+
+    QMap<int, const mixxx::SeratoMarkers2CueEntry*> cueMap;
+    for (auto& pEntry : m_entries) {
+        DEBUG_ASSERT(pEntry);
+        switch (pEntry->typeId()) {
+        case SeratoMarkers2Entry::TypeId::Cue: {
+            const mixxx::SeratoMarkers2CueEntry* pCueEntry = static_cast<mixxx::SeratoMarkers2CueEntry*>(pEntry.get());
+            cueMap.insert(pCueEntry->getIndex(), pCueEntry);
+            break;
+        }
+        case SeratoMarkers2Entry::TypeId::Bpmlock: {
+            const mixxx::SeratoMarkers2BpmlockEntry* pBpmlockEntry = static_cast<mixxx::SeratoMarkers2BpmlockEntry*>(pEntry.get());
+            track->setBpmLocked(pBpmlockEntry->isLocked());
+            break;
+        }
+        // TODO: Add support for COLOR/LOOP/FLIP
+        default:
+            break;
+        }
+    }
+
+    const double sampleRateKhz = track->getSampleRate() / 1000.0;
```

Comment:
Fetching the sample rate from the Track object is wrong. Instead you should pass it separately, e.g. as a member of some "context" object/struct that contains all required parameters.

Reply:
Don't I need the track object anyway to add/remove cues?

Reply:
Unfortunately all cues are heavy weight QObject UI objects with connections, so I suppose yes. 

These should better be split into data objects with a QObject wrapper around them. Like the`TrackInfo`/`AlbumInfo` structs that I extracted in a first attempt from `Track`. For the cues we need a vector of single cues that might be wrapped by a single `TrackCues` object. Connecting and reconnecting individual cues with signals to UI elements was not a good idea as considering the complexity of the resulting code.